### PR TITLE
fix #174 - added `retry when fail` switch to schedule form

### LINKED ISSUE #174: Scheduler - add „retry when fail” checkbox. Default checked

### PR DESCRIPTION
**Pull request by kaspi83**

### CODE REVIEW

File: app/DoctrineMigrations/Version20180116184415.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+class Version20180116184415 extends AbstractMigration
+{
```
Comment (fracz) [inline app/DoctrineMigrations/Version20180116184415.php]: Format this file according to our rules

File: app/DoctrineMigrations/Version20180116184415.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
```
Comment (fracz) [inline app/DoctrineMigrations/Version20180116184415.php]: Describe breifly what this migraiton does

File: src/Frontend/src/schedules/schedule-list/schedule-list.vue [hunk 3]
```diff
@@ -94,6 +99,11 @@
             formatDate(date) {
                 return date ? moment(date).format('LLL') : '-';
             },
+            showState(state) {
+                if (state) {
+                    return `<span class="label label-success">${this.$t('ENABLED')}</span>`;
+                }
```
Comment (fracz) [inline src/Frontend/src/schedules/schedule-list/schedule-list.vue]: else what? empty cell?

File: src/SuplaBundle/Entity/Schedule.php [hunk 4]
```diff
@@ -115,6 +115,12 @@ class Schedule {
      */
     private $caption;
 
+    /**
+     * @ORM\Column(name="retry", type="boolean", nullable=false)
```
Comment (fracz) [inline src/SuplaBundle/Entity/Schedule.php]: Add default=1 here so the schema is updated accordingly. The default assignment below is not sufficient.

File: src/SuplaBundle/Entity/Schedule.php [hunk 5]
```diff
@@ -115,6 +115,12 @@ class Schedule {
...
+     * @Groups({"basic"})
+     */
+    protected $retry = true;
```
Comment (fracz) [inline src/SuplaBundle/Entity/Schedule.php]: Don't you think `retryOnFailure` would be clearer?

File: src/SuplaBundle/Entity/Schedule.php [hunk 6]
```diff
@@ -277,4 +284,18 @@ public function setCaption($caption) {
     public function getUserTimezone() {
         return new \DateTimeZone($this->getUser()->getTimezone());
     }
+
+    /**
+     * @return mixed
+     */
+    public function getRetry() {
```
Comment (fracz) [inline src/SuplaBundle/Entity/Schedule.php]: add return type `:boolean`, delete comment

File: src/SuplaBundle/Entity/Schedule.php [hunk 7]
```diff
@@ -277,4 +284,18 @@ public function setCaption($caption) {
...
+        return $this->retry;
+    }
+
+    /**
+     * @param mixed $retry
+     */
+    public function setRetry($retry) {
```
Comment (fracz) [inline src/SuplaBundle/Entity/Schedule.php]: add typing, delete comment

File: src/SuplaBundle/Enums/ChannelFunction.php [hunk 8]
```diff
@@ -106,20 +106,20 @@ public function getPossibleActions(): array {
 
     public static function actions(): array {
         return [
-            self::CONTROLLINGTHEGATEWAYLOCK => [ChannelFunctionAction::OPEN()],
-            self::CONTROLLINGTHEDOORLOCK => [ChannelFunctionAction::OPEN()],
-            self::CONTROLLINGTHEGATE => [ChannelFunctionAction::OPEN(), ChannelFunctionAction::CLOSE()],
-            self::CONTROLLINGTHEGARAGEDOOR => [ChannelFunctionAction::OPEN(), ChannelFunctionAction::CLOSE()],
+            self::CONTROLLINGTHEGATEWAYLOCK => [ChannelFunctionAction::OPEN],
```
Comment (fracz) [inline src/SuplaBundle/Enums/ChannelFunction.php]: this is likely to break channel groups panel. Generally, we prefer enums than integers as they contain more data. Schedules form should be adjusted to make use of enums, not the other way.

File: src/SuplaDeveloperBundle/DataFixtures/ORM/DevicesFixture.php [hunk 9]
```diff
@@ -90,6 +90,9 @@ private function createDevice(string $name, Location $location, array $channelTy
                 'type' => $channelData[0],
                 'function' => $channelData[1],
                 'channelNumber' => $channelNumber++,
+                'param1' => 0,
```
Comment (fracz) [inline src/SuplaDeveloperBundle/DataFixtures/ORM/DevicesFixture.php]: And this should be another commit as it is not related to the issue

File: src/SuplaBundle/Entity/Schedule.php (hunk 5 above)
Comment (kaspi83) [reply src/SuplaBundle/Entity/Schedule.php]: Refer to the last comment https://github.com/SUPLA/supla-cloud/issues/174#issuecomment-357701024

**Pull request merged**